pkg/reconciler/delivery/delivery.go

```diff
@@ -140,6 +133,21 @@ func (c *Reconciler) fetchRevisions(cfg *v1.Configuration) (map[string]*v1.Revis
 	return revisionMap, nil
 }
 
+// fetchPolicy queries the indexer to retrieve a Policy object and return its translated version
+// if annotations don't specify a Policy or if the specified Policy cannot be found, an error is returned
+func (c *Reconciler) fetchPolicy(cfg *v1.Configuration) (*Policy, error) {
+	// we only need to check the annotations of cfg because owner ksvc's annotation value is propagated
+	policyName, ok := cfg.Annotations[PolicyNameKey]
+	if !ok {
+		return nil, fmt.Errorf("rollout policy is unspecified for %v", cfg.Namespace+"/"+cfg.Name)
```

Comment:
Should this throw an error, or should this do nothing if there's no policy?

Reply:
Throwing an error is a conscious decision here because I imagined that someone (like me) might forget to annotate their Service/Configuration and need a friendly reminder from the logs. If the person didn't wanna use any policy in the first place then everything just directly jumps to 100% and he/she didn't have to bother with the whole KCD business in the first place.

Reply:
I can see how forgetting to add a policy means that your new revision is now rolled out 100% which is bad. However, not allowing users to opt-out of progressive rollout is also IMO a bad user experience. IMHO, the concern of "ensure a policy is set" and "enforce the rollout policy" are not the same concern and should be handled by different systems.

I see a few alternatives for enforcing policy:
1. We could add a config-delivery.yaml configmap that specifies "require-policy" field, which determines if a policy needs to be added on a service or if it could be omitted. This could be a validating webhook. This is useful because we anticipate this type of policy enforcement, but is limited to "policy exists or not".
2. A user could also implement this validating webhook if they want to enforce policies. 
3. A user could use a system like Gatekeeper to enforce that the label is always set: https://github.com/open-policy-agent/gatekeeper#how-to-use-gatekeeper. This is the most robust and flexible because it would allow platform operators to set even more specific requirements on the policy such as "it must be this one specific policy in the prod namespace, but it could be any of these 4 policies in the staging namespace." Or "Team A has these policy options and Team B has these other policy options".

Thoughts?